File: app/src/main/java/xyz/rodit/snapmod/features/tweaks/LocationOverride.java
package xyz.rodit.snapmod.features.tweaks;

import de.robv.android.xposed.XC_MethodHook;
import xyz.rodit.snapmod.features.Feature;
import xyz.rodit.snapmod.features.FeatureContext;
import xyz.rodit.snapmod.mappings.LocationMessage;
import xyz.rodit.snapmod.mappings.LocationMessageBuilder;

public class LocationOverride extends Feature {

    public LocationOverride(FeatureContext context) {
        super(context);
    }

    @Override
    protected void performHooks() {
        // Override location (not sure if this works)
        LocationMessageBuilder.transform.hook(new XC_MethodHook() {
            @Override
            protected void beforeHookedMethod(MethodHookParam param) {
                if (context.config.getBoolean("location_share_override")) {
                    double lat = Double.parseDouble(context.config.getString("location_share_lat", "0"));
                    double $long = Double.parseDouble(context.config.getString("location_share_long", "0"));
                    LocationMessage loc = LocationMessage.wrap(param.args[0]);
                    loc.setLatitude(lat);
                    loc.setLongitude($long);
                }
            }
        });
    }
}

File: app/src/main/java/xyz/rodit/snapmod/features/tweaks/LocationOverride.kt
package xyz.rodit.snapmod.features.tweaks

import xyz.rodit.snapmod.features.Feature
import xyz.rodit.snapmod.features.FeatureContext
import xyz.rodit.snapmod.mappings.LocationMessage
import xyz.rodit.snapmod.mappings.LocationMessageBuilder
import xyz.rodit.snapmod.util.before

class LocationOverride(context: FeatureContext) : Feature(context) {

    override fun performHooks() {
        // Override location (not sure if this works).
        LocationMessageBuilder.transform.before(context, "location_share_override") {
            val lat = context.config.getString("location_share_lat", "0").toDouble()
            val long = context.config.getString("location_share_long", "0").toDouble()
            val loc = LocationMessage.wrap(it.args[0])
            loc.latitude = lat
            loc.longitude = long
        }
    }
}